dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLlvm.java

```diff
@@ -859,6 +859,14 @@ public Expression visitArrayConst(ArrayConstContext ctx) {
         return expressions.makeArray(elementType, arrayValues, true);
     }
 
+    @Override
+    public Expression visitPoisonConst(PoisonConstContext ctx) {
+        // It is correct to replace a poison value with an undef value or any value of the type.
+        BooleanType booleanType = types.getBooleanType();
+        var nondeterministicExpression = new BNonDet(booleanType);
+        return expressions.makeCast(nondeterministicExpression, expectedType);
```

Comment:
Why do you always use a boolean type? I think you should use the correct type. 
Btw. I think it would also be fine to use a constant value like `0` (we do this for `undef`).